### PR TITLE
sys: fmt: add fmt_u64_dec(), print_u64_dec()

### PR DESCRIPTION
**Pull request by kaspar030**
This PR adds previously missing uin64_t decimal fmt/print to the fmt module.

It *does* add >400bytes of code to SAM R21 *edit* (when used) *edit*, but that is still a lot less than \*printf with 64bit support will take.
~~And *please* find a less clumsy way to add up the base 10k parts...~~

### CODE REVIEW

**Comment by Kijewski** [comment]
Hm, `fmt_u64_dec()` is incredibly ugly. At this point is there really an advantage over a simple mod-divide loop?

**Comment by kaspar030** [comment]
> Hm, fmt_u64_dec() is incredibly ugly. At this point is there really an advantage over a simple mod-divide loop?

Using mod-divide is ~170 bytes larger, as it pulls 64bit division stuff. So unless that is used somewhere else, it's maybe worth it... Sadly we cannot automatically use one or the other depending on whether that is the case.

**Comment by kaspar030** [comment]
I've made the final printing (IMHO) less ugly (and carved off ~30bytes).

**Comment by Kijewski** [comment]
Yeah, looks less like dark magic now. :) Maybe add a test to unittests/tests-fmt?

**Comment by OlegHahm** [comment]
I think a test could go into a separate PR.

**Comment by kaspar030** [comment]
Well, simple to copy&paste some. ;)

**Comment by kaspar030** [comment]
- rebased
- squashed

@cgundogan Hey, this is a low-hanging fruit. :)

**Comment by cgundogan** [comment]
>@cgundogan Hey, this is a low-hanging fruit. :)

Is it? I just tested and it works correctly. Code looks also good now. Let's squeeze these vitamins out of that low hanging fruit!

ACK and GO once travis is green

File: sys/fmt/fmt.c [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include <stdarg.h>
 #include <stdint.h>
 #include <unistd.h>
+#include <string.h>
```
Comment (cgundogan) [inline sys/fmt/fmt.c]: I am wondering, what is string.h needed for?

Comment (cgundogan) [inline sys/fmt/fmt.c]: Oh, probably for memset?

Comment (kaspar030) [inline sys/fmt/fmt.c]: yep

**Comment by kaspar030** [comment]
thx! :)

**Pull request merged**